generator/generator_tests/road_access_test.cpp

```diff
@@ -0,0 +1,130 @@
+#include "testing/testing.hpp"
+
+#include "generator/generator_tests_support/routing_helpers.hpp"
+#include "generator/generator_tests_support/test_feature.hpp"
+#include "generator/generator_tests_support/test_mwm_builder.hpp"
+
+#include "generator/road_access_generator.hpp"
+
+#include "routing/road_access_serialization.hpp"
+
+#include "indexer/index.hpp"
+#include "indexer/mwm_set.hpp"
+
+#include "coding/file_name_utils.hpp"
+
+#include "platform/platform_tests_support/scoped_dir.hpp"
+#include "platform/platform_tests_support/scoped_file.hpp"
+
+#include "platform/country_file.hpp"
+#include "platform/platform.hpp"
+
+#include "base/logging.hpp"
+
+#include "std/string.hpp"
+
+using namespace feature;
+using namespace generator;
+using namespace platform::tests_support;
+using namespace platform;
+using namespace routing;
+
+namespace
+{
+string const kTestDir = "road_access_generation_test";
+string const kTestMwm = "test";
+string const kRoadAccessFilename = "road_access_in_osm_ids.csv";
+string const kOsmIdsToFeatureIdsName = "osm_ids_to_feature_ids" OSM2FEATURE_FILE_EXTENSION;
+
+void BuildEmptyMwm(LocalCountryFile & country)
+{
+  generator::tests_support::TestMwmBuilder builder(country, feature::DataHeader::country);
+}
+
+void LoadRoadAccess(MwmValue const & mwmValue, RoadAccess & roadAccess)
+{
+  TEST(mwmValue.m_cont.IsExist(ROAD_ACCESS_FILE_TAG), ());
+
+  try
+  {
+    FilesContainerR::TReader const reader = mwmValue.m_cont.GetReader(ROAD_ACCESS_FILE_TAG);
```

Comment:
So, you use only m_cont from MwmValue. Do you really need MwmValue in this case? Why not to directly use FileContainerR, and you may get rid of Index.